odoo/tools/translate.py

```diff
@@ -1043,8 +1043,10 @@ def trans_load_data(cr, fileobj, fileformat, lang, lang_name=None, verbose=True,
 
             # Make a reader for the POT file and be somewhat defensive for the
             # stable branch.
-            if fileobj.name.endswith('.po'):
-                try:
+            try:
+                # changed 'if' inside 'try', a tempfile set name to its fd (e.g, 7)
+                # in python3 (instead of a str with value '<fopen>' in python2)
+                if fileobj.name.endswith('.po'):
```

Comment:
I wouldn't inverse this. Just make sure to make it compatible with Python 3. You might have to use `typeof` to avoid crashes.